### PR TITLE
[MERGE] base: remove read access to ir.ui.view

### PR DESCRIPTION
**Pull request by mart-e**
Only system and designer users can access `ir.ui.view` records.
An employee, portal or public user should not be able to read a view arch, while
ok in standard (all views are in the code anyway), one may want to integrate
confidential business logic in some views.
All operations on views must be done in `sudo` or use helpers like
`fields_view_get` or `_render`.

This PR is part of the long-term task id 8203 that plans to remove all access to
`ir.*` models

**Changes of ACL**:
- remove read for all on `ir.ui.view`
- remove read for all on `website.page` (uses `_inherits`)
- add read on `ir.ui.view` for website publisher (cf `read_template`)
(maintain all for system and designer)

**Changes of method**:
- `fields_view_get` uses `sudo`. Can only be called only if we have the read
  access on the linked model.
- `render*` becomes private `_render*` and uses `sudo`. Shouldn't be able to
  arbitraly render any template. A QWeb template is not linked to a model so
  can't apply the same ACL check as in `fields_view_get`.
- `read_templates` remains public but without `sudo`. It is called by the method
  `load` method (now private too) or by the web_editor module

**Changes of behaviour**
- the `groups_id` parameter may be used on a view to relax the access and allow
  a specific group to access its content (e.g. frontend assets for employees)

### CODE REVIEW

File: addons/website/tests/test_performance.py [hunk 1]
```diff
@@ -77,25 +77,25 @@ def _create_page_with_menu(self, url):
 
     def test_10_perf_sql_queries_page(self):
         # standard untracked website.page
-        self.assertEqual(self._get_url_hot_query(self.page.url), 11)
+        self.assertEqual(self._get_url_hot_query(self.page.url), 18)
```
Comment (JKE-be) [inline addons/website/tests/test_performance.py:80]: :fearful:  ?? Seriously ?

File: addons/website/models/ir_ui_view.py [hunk 2]
```diff
@@ -359,10 +359,7 @@ def _handle_visibility(self, do_raise=True):
         """
         error = False
 
-        try:
-            self.visibility  # avoid useless sudo() in case page is public
-        except AccessError:
-            self = self.sudo()
+        self = self.sudo()
```
Comment (JKE-be) [inline addons/website/models/ir_ui_view.py:362]: :heart:

File: addons/website/models/website.py [hunk 3]
```diff
@@ -670,7 +670,7 @@ def viewref(self, view_id, raise_if_not_found=True):
             :param raise_if_not_found: should the method raise an error if no view found
             :return: The view record or empty recordset
         '''
-        View = self.env['ir.ui.view']
+        View = self.env['ir.ui.view'].sudo()
```
[the remark this answers is not in the document]
Comment (JKE-be) [reply addons/website/models/website.py:673]: Yes it is planned here:
https://github.com/odoo-dev/odoo/commit/8212e51f376602866873067629cc4764910d6385

wait a review with Al before merging, but should be done asap

[the remark this answers is not in the document]
Comment (JKE-be) [reply addons/website/models/website.py:673]: Shoud be adapted, for now it is only valid for show_customize view.
Maybe we should just make it more generic.

File: addons/website/models/website.py [hunk 4]
```diff
@@ -833,7 +833,7 @@ def enumerate_pages(self, query_string=None, force=False):
         if query_string:
             domain += [('url', 'like', query_string)]
 
-        pages = self.get_website_pages(domain)
+        pages = self.sudo().get_website_pages(domain)
```
[the remark this answers is not in the document]
Comment (mart-e) [reply addons/website/models/website.py:836]: if private, the sudo can be moved inside then, fine with me

File: addons/website/security/ir.model.access.csv [hunk 5]
```diff
@@ -5,9 +5,10 @@ access_website_menu,access_website_menu,model_website_menu,,1,0,0,0
 access_website_menu_designer,Web Menu Manager,model_website_menu,group_website_designer,1,1,1,1
 access_website_rewrite,access_website_rewrite,model_website_rewrite,,0,0,0,0
 access_website_rewrite_designer,Web Rewrite Manager,model_website_rewrite,group_website_designer,1,1,1,1
-access_website_page,access_website_page,model_website_page,,1,0,0,0
+access_website_page,access_website_page,model_website_page,,0,0,0,0
```
[the remark this answers is not in the document]
Comment (mart-e) [reply addons/website/security/ir.model.access.csv:8]: The read on ir.ui.view is to be able to load the definition of the website snippets (which are added by themes). Not for the actual website pages (which are retrieved via the controllers like any other page)

File: odoo/addons/base/models/ir_ui_view.py [hunk 6]
```diff
@@ -600,6 +600,23 @@ def accessible(view):
 
         return self.browse(view_ids).sudo().filtered(accessible)
 
+    def _accessible(self, raise_error=False):
```
[the remark this answers is not in the document]
Comment (mart-e) [reply odoo/addons/base/models/ir_ui_view.py:603]: actually, I realised all the calls are with `raise_error=True` so I can probably just drop this parameter

**Comment by mart-e** [comment]
@robodoo r+ merge

**Pull request merged**